File: SPX.DevTools/DevConsole.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SPX.Text;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace SPX.Core
{
    public class DevConsole
    {
        private class ConsoleText : SPX.Text.Text
        {
            public ConsoleText(int x, int y, string content)
                : base(content, x, y, TextType.Action)
            {

            }

            public override void Draw()
            {                
                XnaManager.Renderer.DrawString(TextManager.GetFont(), _contents, _position, Color.Black, 0, Vector2.Zero, 1.15f, SpriteEffects.None,.9994f);
                XnaManager.Renderer.DrawString(TextManager.GetFont(), _contents, _position, Color.White, 0, Vector2.Zero, 1.0f, SpriteEffects.None, .9995f);
            }
        }

        private static DevConsole __instance;
        public static DevConsole Get()
        {
            if (__instance == null)
            {
                __instance = new DevConsole();
            }
            return __instance;
        }

        private ConsoleText[] _contents = new ConsoleText[10];
        private int _index = 0;
        private bool _isVisible;
        private Color _bgColor;

        private DevConsole()
        {
            _bgColor = Color.Black;
            _bgColor.A = (byte)180;
            Add("The console has been started.");
        }

        public void Add(string message)
        {
            _contents[++_index] = new ConsoleText(50,_index*50,message);
        }

        public void Draw()
        {
            if (_isVisible)
            {
                XnaManager.Renderer.Draw(XnaManager.GetMenuBaseAsset(), Vector2.Zero, new Rectangle(0, 0, XnaManager.WindowWidth, XnaManager.WindowHeight), _bgColor, 0f, Vector2.Zero, 1, SpriteEffects.None, .9993f);
                for (int ii = 0; ii < _contents.Length; ii++)
                {
                    if (_contents[ii] != null)
                    {
                        _contents[ii].Draw();
                    }
                }
            }
        }

        public void Toggle()
        {
            _isVisible = !_isVisible;
        }
    }
}

File: proj/src/spx/devtools/DevConsole.java
package spx.devtools;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.Sprite;
import spx.bridge.DrawDepth;
import spx.core.Point2;
import spx.core.SpxManager;
import spx.text.TextType;

public class DevConsole {

    public static final int margin = 50;

    private class ConsoleText extends spx.text.Text {
        public ConsoleText(int x, int y, String content) {
            super(content, x, y, TextType.Action);
        }

        @Override
        public void draw() {
            SpxManager.Renderer.drawString(_contents, _position, Color.WHITE, 1.0f, DrawDepth.DevConsoleText);
        }

        public String getContent() {
            return _contents;
        }
    }

    private static DevConsole __instance;

    public static DevConsole get() {
        if (__instance == null) {
            __instance = new DevConsole();
        }
        return __instance;
    }

    private final ConsoleText[] _contents = new ConsoleText[10];
    private int _index = 0;
    private boolean _isVisible = false;
    private final Color _bgColor;
    private final Sprite _consoleBase;

    private DevConsole() {
        _bgColor = Color.BLACK;
        _bgColor.a = (byte) 180;
        _consoleBase = SpxManager.getMenuBaseAsset();
        add("The development console has been started.");
    }

    private int getY(int index) {
        return SpxManager.WindowHeight - (index * margin / 2);
    }

    public void add(String message) {
        if (_index < _contents.length) {
            _contents[_index++] = new ConsoleText(margin, getY(_index), message);
        } else {
            for (int ii = 0; ii < _contents.length - 1; ii++) {
                _contents[ii] = new ConsoleText(margin, getY(ii), _contents[ii + 1].getContent());
            }
            _contents[_contents.length - 1] = new ConsoleText(margin, getY(_contents.length - 1), message);
        }
    }

    public void draw() {
        if (_isVisible) {
            SpxManager.Renderer.draw(_consoleBase, Point2.Zero, DrawDepth.DevConsole, _bgColor, SpxManager.WindowWidth, SpxManager.WindowHeight);
            for (int ii = 0; ii < _contents.length; ii++) {
                if (_contents[ii] != null) {
                    _contents[ii].draw();
                }
            }
        }
    }

    public void toggle() {
        _isVisible = !_isVisible;
    }
}
